### PR TITLE
Add log while not using user input

### PR DESCRIPTION
**Pull request by rickwu666666**
* Add log while using default timezone.

2024-06-12 16:04:05,410 [INFO] Start to test timezone setting up
2024-06-12 16:04:05,639 [INFO] Current timezone is the same as target timezone Asia/Taipei
2024-06-12 16:04:05,639 [INFO] Will use default time zone UTC
2024-06-12 16:04:05,639 [INFO] Attempting to set timezone to UTC
2024-06-12 16:04:05,726 [INFO] Current timezone is UTC
2024-06-12 16:04:05,727 [INFO] Timezone setup succeed!
2024-06-12 16:04:05,727 [INFO] Restoring original timezone to Asia/Taipei
2024-06-12 16:04:05,727 [INFO] Attempting to set timezone to Asia/Taipei
2024-06-12 16:04:05,884 [INFO] Current timezone is Asia/Taipei
2024-06-12 16:04:05,884 [INFO] Timezone setup succeed!

* Add unittest

### CODE REVIEW

File: bin/timedatectl_test.py [hunk 1]
```diff
@@ -201,6 +198,13 @@ def test_timezone(target_timezone):
         with TimeZoneRestore() as tzr:
             if tzr.restore_timezone in timezone_list:
                 timezone_list.remove(tzr.restore_timezone)
+                if tzr.restore_timezone == target_timezone:
+                    logging.info(
+                        "Current timezone is the same as target timezone %s",
+                        target_timezone,
+                    )
+                    logging.info("Will use default time zone %s",
+                                 timezone_list[-1])
```
Comment (stanley31huang) [inline bin/timedatectl_test.py:207]:
```suggestion
                    logging.info(
                        "Current timezone is the same as target timezone, "
                        "so the timezone %s is used instead.",
                        timezone_list[-1],
                    )
```

File: bin/timedatectl_test.py [hunk 2]
```diff
@@ -54,13 +54,10 @@ def timedatectl_parser() -> Dict[str, str]:
         LOCAL_TIME: None,
     }
     output = run("timedatectl")
-    timezone_search = re.search(
-        r"Time zone:\s+(\S+)\s+\(.+\)", output
-    )
+    timezone_search = re.search(r"Time zone:\s+(\S+)\s+\(.+\)", output)
```
Comment (stanley31huang) [inline bin/timedatectl_test.py:57]:
```suggestion
    timezone_search = re.search(r"Time zone: (\S+) \(.+\)", output)
```

**Comment by rickwu666666** [comment]
@stanley31huang I've pushed the fix. Many thanks for the review.

**Comment by stanley31huang** [comment]
Per discussion with Rick, we will drop this PR and refactor the scripts first.
https://github.com/canonical/test-strict-confinement/issues/6

**Pull request closed without merging**